cmd/swarm/main.go

```diff
@@ -491,34 +454,78 @@ func registerBzzService(ctx *cli.Context, stack *node.Node) {
 			}
 		}
 
-		var ensClient *ethclient.Client
-		if ensapi != "" {
-			log.Info("connecting to ENS API", "url", ensapi)
-			client, err := rpc.Dial(ensapi)
-			if err != nil {
-				return nil, fmt.Errorf("error connecting to ENS API %s: %s", ensapi, err)
+		ensClientConfigs := []swarm.ENSClientConfig{}
+		switch len(ensAPIs) {
+		case 0:
+			ensClientConfigs = append(ensClientConfigs, swarm.ENSClientConfig{
+				Endpoint:        node.DefaultIPCEndpoint("geth"),
+				ContractAddress: ensAddr,
+			})
+		case 1:
+			// Check if only one --ens-api is specified in order to use --ens-addr value
+			// to preserve the backward compatibility with single --ens-api flag.
+			// Multiple
+			c := parseFlagEnsAPI(ensAPIs[0])
```

Comment:
We currently treat `--ens-api ''` as an indication to disable the use of ENS so we should check for that case here (but to be honest, we should have an explicit flag like `--no-ens` instead).